### PR TITLE
Cleanup it tests

### LINKED ISSUE #34: Cleanup it-tests
**Motivation**
IT tests uses currently example templates. Markup is big and it is often hard to understand what happens and what actually is tested.

**Describe the solution you'd like**
`knotx-it-tests` example content files and results be simplified.
Files should contain only the thing that is tested in particular unit test.
There may be one big file to test multiple things together.

See some @tomaszmichalak comments from #33

### PR DESCRIPTION
**Pull request by Skejven**
## Description
Fixed #34 .

## Motivation and Context
IT tests contained a lot of unnecessary and some tests that made them hard to understand.
Now they are cleaned up.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-it-tests/src/test/resources/conf/it-tests-application.conf [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  "httpRepo=io.knotx.repository.http.HttpRepositoryConnectorVerticle"
+  "fsRepo=io.knotx.repository.fs.FilesystemRepositoryConnectorVerticle"
+  "splitter=io.knotx.splitter.FragmentSplitterVerticle"
+  "assembler=io.knotx.assembler.FragmentAssemblerVerticle"
+  ########### Data Bridge ###########
+  "dataBridge=io.knotx.databridge.core.DataBridgeKnot"
+  "dataSourceHttp=io.knotx.databridge.http.HttpDataSourceAdapter"
+  ########### Forms ###########
+  "forms=io.knotx.forms.core.FormsKnot"
+  ########### Template Engine ###########
```
Comment (tomaszmichalak) [inline knotx-it-tests/src/test/resources/conf/it-tests-application.conf]: Please add here the comment that Forms adapter is defined and initialized in Test class.

File: knotx-it-tests/src/test/resources/content/local/customSnippetTag.html [hunk 2]
```diff
@@ -17,71 +17,34 @@
...
-        <p>
-          <a class="btn btn-primary btn-large"
-             href="https://github.com/Cognifide/knotx">Learn more</a>
-        </p>
-      </div>
-    </div>
-  </div>
-  <div class="row">
+  <div>
     <cusotm:snippet it-test-knots="databridge,te"
```
Comment (tomaszmichalak) [inline knotx-it-tests/src/test/resources/content/local/customSnippetTag.html]: Maybe <custom:snippet> :) ?

**Comment by tomaszmichalak** [comment]
Good job!

File: knotx-it-tests/src/test/java/io/knotx/example/monolith/SampleApplicationIntegrationTest.java [hunk 3]
```diff
@@ -89,107 +81,92 @@ void initMocks() {
...
       VertxTestContext context, Vertx vertx) {
-    testGetRequest(context, vertx, LOCAL_MULTIPLE_FORMS_URI, "multipleFormWithGetResult.html");
+    testGetServerError(context, vertx, "/content/local/notExistingService.html");
   }
 
   @Test
-  @KnotxApplyConfiguration("knotx-test-app.conf")
-  public void whenRequestingWithPostMethodFirstForm_expectFirstFormPresentingFormActionResult(
+  @KnotxApplyConfiguration("conf/it-tests-application.conf")
+  public void requestPageWithMissingDataDefinitionAndFallbackDefined(
```
[the remark this answers is not in the document]
Comment (Skejven) [reply knotx-it-tests/src/test/java/io/knotx/example/monolith/SampleApplicationIntegrationTest.java]: Renamed

File: knotx-it-tests/src/test/java/io/knotx/stack/it/KnotxServerTester.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  void testPostRequest(VertxTestContext context, Vertx vertx, String url,
+      Map<String, Object> formData, String expectedResponseFile) {
+
+    WebClient client = WebClient.create(vertx);
+    Single<HttpResponse<Buffer>> httpResponseSingle = client
+        .post(serverPort, serverHost, url)
+        .rxSendForm(getMultiMap(formData));
+
+    subscribeToResult_shouldSucceed(context, httpResponseSingle, resp -> {
+      assertEquals(Jsoup.parse(FileReader.readTextSafe(expectedResponseFile)).body().html(),
```
[the remark this answers is not in the document]
Comment (Skejven) [reply knotx-it-tests/src/test/java/io/knotx/stack/it/KnotxServerTester.java]: Done

[the remark this answers is not in the document]
Comment (Skejven) [reply knotx-it-tests/src/test/java/io/knotx/stack/it/KnotxServerTester.java]: Good idea.
However, I will create assertion that checks HTML body.

[the remark this answers is not in the document]
Comment (Skejven) [reply knotx-it-tests/src/test/java/io/knotx/stack/it/KnotxServerTester.java]: See https://github.com/Knotx/knotx-junit5/pull/26

**Pull request merged**